cmd/kubelet/app/options/options.go

```diff
@@ -434,7 +435,9 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 
 	fs.StringVar(&c.VolumePluginDir, "volume-plugin-dir", c.VolumePluginDir, "The full path of the directory in which to search for additional third party volume plugins")
 	fs.StringSliceVar(&c.ClusterDNS, "cluster-dns", c.ClusterDNS, "Comma-separated list of DNS server IP address.  This value is used for containers DNS server in case of Pods with \"dnsPolicy=ClusterFirst\". Note: all DNS servers appearing in the list MUST serve the same set of records otherwise name resolution within the cluster may not work correctly. There is no guarantee as to which DNS server may be contacted for name resolution.")
-	fs.DurationVar(&c.StreamingConnectionIdleTimeout.Duration, "streaming-connection-idle-timeout", c.StreamingConnectionIdleTimeout.Duration, "Maximum time a streaming connection can be idle before the connection is automatically closed. 0 indicates no timeout. Example: '5m'. Note: All connections to the kubelet server have a maximum duration of 4 hours.")
+	dummyDuration := 4 * time.Hour
+	fs.DurationVar(&dummyDuration, "streaming-connection-idle-timeout", dummyDuration, "Maximum time a streaming connection can be idle before the connection is automatically closed. 0 indicates no timeout. Example: '5m'. Note: All connections to the kubelet server have a maximum duration of 4 hours.")
+	fs.MarkDeprecated("streaming-connection-idle-timeout", "This flag has no effect and will be removed in a future version.") //nolint:errcheck
```

Comment:
I was checking previous precedent , it seems  is ok to remove after some time announcing the deprecation https://github.com/kubernetes/kubernetes/pull/114446/commits/556d3fc2b7b4dfd1d94d3039909ddb42d88f7c52

I defer to node owners to decide how to better do it, but I suggest to keep at least one release of warning

maybe 
```suggestion
	fs.MarkDeprecated("streaming-connection-idle-timeout", "This flag has no effect and will be removed in v1.31.") //nolint:errcheck
```
??

Reply:
Changed as suggested.